nhrpd/nhrp_cache.c

```diff
@@ -119,12 +119,47 @@ static void nhrp_cache_update_route(struct nhrp_cache *c)
 {
 	struct prefix pfx;
 	struct nhrp_peer *p = c->cur.peer;
+	char buf[3][SU_ADDRSTRLEN];
+	struct nhrp_interface *nifp;
 
 	sockunion2hostprefix(&c->remote_addr, &pfx);
 
 	if (p && nhrp_peer_check(p, 1)) {
-		netlink_update_binding(p->ifp, &c->remote_addr,
-				       &p->vc->remote.nbma);
+		if (sockunion_family(&c->cur.remote_nbma_natoa) != AF_UNSPEC) {
+			/* remote_nbma_natoa is already set. Therefore, binding
```

Comment:
Could you explain why? The NAT O-A is basically the private address of the device.  This breaks the scenario when two devices are behind NAT, but the other one is having port forwarding of public to private address.

Maybe the intention was to fix when two devices are behind the same NAT device. But this is not supported due to other reasons.

This should be revert.